internal/configs/static_evaluator.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package configs
+
+import (
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/opentofu/opentofu/internal/addrs"
+	"github.com/opentofu/opentofu/internal/lang"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// StaticIdentifier holds a Referencable item and where it was declared
+type StaticIdentifier struct {
+	Module    addrs.Module
+	Subject   string
+	DeclRange hcl.Range
+}
+
+func (ref StaticIdentifier) String() string {
+	val := ref.Subject
+	if len(ref.Module) != 0 {
+		val = ref.Module.String() + ":" + val
+	}
+	return val
+}
+
+type StaticModuleVariables func(v *Variable) (cty.Value, hcl.Diagnostics)
+
+// StaticModuleCall contains the information required to call a given module
+type StaticModuleCall struct {
+	addr      addrs.Module
+	vars      StaticModuleVariables
+	rootPath  string
+	workspace string
+}
+
+func NewStaticModuleCall(addr addrs.Module, vars StaticModuleVariables, rootPath string, workspace string) StaticModuleCall {
+	return StaticModuleCall{
+		addr:      addr,
+		vars:      vars,
+		rootPath:  rootPath,
+		workspace: workspace,
+	}
+}
+
+//nolint:gochecknoglobals // only used in testing
+var RootModuleCallForTesting = StaticModuleCall{
```

Comment:
nit: let's move it to `_test.go` if it's only used in testing

also, it could be a function instead of a global var so we don't need to nolint

Reply:
Makes sense. What about making it a function to remove global state?